src/Auth.js

```diff
@@ -203,10 +203,13 @@ Auth.prototype.getRolesForUser = function() {
     return query.execute().then(({ results }) => results);
```

Comment:
Got it !

Reply:
As an example there is the main logic of the Parse.Query.each there: https://github.com/parse-community/Parse-SDK-JS/blob/master/src/ParseQuery.js#L722

basically:

1. force order on objectId ascending (safe as indexed)
2. if the last result has exactly `limit` results
   - use the last object'd object ID as a constraint (where objectId > lastObjectId)
    - run the query with the new constraints and go back to 2

Reply:
Do you know if it's possible to reproduce the `.each()` behavior directly with `RestQuery` ? Or it need some extra code for iterate ?

Reply:
RestQueey will just return you results, you’ll need the code, similar to what I shared above, most precisely the continueWhile part. Gimme à minute, i’ll Try to give you a quick snippet :) in the meantime, can you check to add a test for this? Creating a user with many roles (200) and ensuring your fix is effective :)

Reply:
a `.each()` like code from ParseQuery.js from Parse-SDK-JS seems a little heavy to me for just a fix of this size :/

Reply:
it's just a few lines :)

Reply:
```
// move a the top
    const { continueWhile } = require('parse/lib/node/promiseUtils');
    // Wether it's finished or not
    let finished = false;
    // Default limit to 100, change to test
    const queryOptions = {
      limit: 100,
      order: 'objectId'
    };
    // ALl the results
    let results = [];
    const masterConfig = master(this.config);
    // This will execute the second function as long as the first on returns false
    await continueWhile(() => {
      return !finished;
    }, async () => {
      // make the query
      const query = new RestQuery(
        this.config,
        masterConfig,
        '_Role',
        restWhere,
        queryOptions
      );
      // get the list of results
      const currentResults = await query.execute().then(({ results }) => results);
      // we are finished when we have less results than the limit
      finished = currentResults.length < queryOptions.limit;
      // Update the where query to get object after
      if (!finished) {
        restWhere.objectId = { '$gt': currentResults[currentResults.length - 1].id}
      }
      results = results.concat(currentResults);
    });
    return results;
```

This should do :)